### ISSUE TITLE
Support numpy 2.0.0

### ISSUE THREAD

**Issue by dennisbader** [unit8co/darts#2414]
**Describe the bug**
Numpy released version 2.0.0. Many of our dependencies are currently broken. We set an upper cap on `numpy<2.0.0` in #2413.

Dependencies explicitly capping `numpy<2.0.0` (obtained using `pipdeptree`)
- [x] contourpy==1.2.0 [requires: numpy>=1.20,<2.0]
- [x] matplotlib==3.8.3 [requires: numpy>=1.21,<2]
- [x] numba==0.59.0 [requires: numpy>=1.22,<1.27]
- [x] pandas==2.2.1 [requires: numpy>=1.23.2,<2]
- [x] pyarrow==15.0.0 [requires: numpy>=1.16.6,<2]
- [x] scikit-learn==1.4.1.post1 [requires: numpy>=1.19.5,<2.0]
- [x] scipy==1.12.0 [requires: numpy>=1.22.4,<1.29.0]
- [x] statsmodels==0.14.1 [requires: numpy>=1.18,<2]
- [x] pmdarima
  - we removed support for it in #2734
- [ ] catboost:
  - seems to be in development, see [here](https://github.com/catboost/catboost/issues/2671)

**Comment by lorentzenchr** [comment]
Numpy 2.0 compatibility of all dependencies is available:
- [contourpy 1.2.1](https://contourpy.readthedocs.io/en/latest/changelog.html#v1-2-1-2024-04-02)
- [matplotlib 3.8.4](https://github.com/matplotlib/matplotlib/releases/tag/v3.8.4)
- [numba 0.60](https://numba.readthedocs.io/en/stable/release/0.60.0-notes.html)
- [pandas 2.2.2](https://pandas.pydata.org/docs/whatsnew/v2.2.2.html#pandas-2-2-2-is-now-compatible-with-numpy-2-0)
- [pyarrow 16.0.0](https://arrow.apache.org/release/16.0.0.html)
- [scikit-learn 1.4.2](https://scikit-learn.org/stable/whats_new/v1.4.html#version-1-4-2)
- [scipy 1.13.0](https://docs.scipy.org/doc/scipy/release/1.13.0-notes.html)
- [statsmodels 0.14.2](https://github.com/statsmodels/statsmodels/releases/tag/v0.14.2)

See also https://github.com/numpy/numpy/issues/26191.

**Comment by furkancanturk** [comment]
Hi @dennisbader 
Do you plan a release supporting Numpy 2?

**Comment by madtoinou** [comment]
Thanks @lorentzenchr for checking this.

Hi @furkancanturk, yes, the point of this issue to make sure all the dependencies are supporting numpy 2 before removing the version cap for darts. However, we are planning on one last release for python 3.8 before tackling this.

**Comment by swertz** [comment]
Be warned that catboost does not explicitly cap its numpy dependency, but does not support numpy 2.0 yet: https://github.com/catboost/catboost/issues/2671

**Comment by felixdivo** [comment]
This might also be a problem: https://github.com/alkaline-ml/pmdarima/issues/577

**Comment by madtoinou** [comment]
We are waiting for the dependencies to support it before propagating to Darts. If it takes too long, we might try to move away from some of them but it's still too early to take such a decision.

Is the cap on numpy causing issues to some of you? If so, would it be possible to know which other librairy used in combination with Darts requires numpy >= 2.0.0?

**Comment by veducha** [comment]
I agree, I've been running into problems with `pmdarima`,. The project has not had any releases for a long time, so it might be a good idea to move away from it.

**Comment by lorentzenchr** [comment]
Maybe th AutoARIMA from https://github.com/Nixtla/statsforecast could be an alternative for pmdarima?

**Comment by madtoinou** [comment]
It's already included in darts and will most certainly become the new default "engine/backend" for the `AutoARIMA` model, we are discussing about how to make this change with @dennisbader.

**Comment by john-ramsey** [comment]
Just dove down the rabbit hole here and looks like catboost just closed out [their issue](https://github.com/catboost/catboost/issues/2671#issuecomment-2799902671)

**Comment by dennisbader** [comment]
That's great news! Thanks for notifying us @john-ramsey 🚀
